test/e2e_node/device_plugin_test.go

```diff
@@ -152,11 +143,16 @@ func testDevicePlugin(f *framework.Framework, pluginSockDir string) {
 			}, 30*time.Second, framework.Poll).Should(gomega.BeTrue())
 
 			ginkgo.By("Creating one pod on node with at least one fake-device")
-			podRECMD := "devs=$(ls /tmp/ | egrep '^Dev-[0-9]+$') && echo stub devices: $devs"
+			// TODO: the sleep here is racy and fragile. What we need here is:
+			// wait long enough for the pod to be reported Ready, BUT wait for a fixed
+			// amount of time so we have restarts - which we check later in the flow.
+			// A probably better approach would be to trigger restart from the outside
+			// (ExecCommandOnContainer?)
+			podRECMD := "devs=$(ls /tmp/ | egrep '^Dev-[0-9]+$') && echo stub devices: $devs && sleep 1m"
```

Comment:
I think it'd probably be better to call the wait from the go code rather than shelling out, I almost completely missed that we added a sleep here.

Reply:
is going to be challenging because this is the entry point of the test container (see makeBusyboxPod), and we need to work with the images we have available.